kafka-connect/kafka-connect/src/test/java/org/apache/iceberg/connect/data/TestIcebergWriterFactory.java

```diff
@@ -96,4 +97,33 @@ public void testAutoCreateTable(boolean partitioned) {
     assertThat(capturedArguments.get(1)).isEqualTo(Namespace.of("foo1", "foo2"));
     assertThat(capturedArguments.get(2)).isEqualTo(Namespace.of("foo1", "foo2", "foo3"));
   }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void testAutoCreateTableWithInvalidPartitionSpec() {
```

Comment:
we are testing an existing functionality and not the change do we wanna mock the logger to check the log message contains the expected change ?